lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,331 @@
+
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 6),
+      child: TextButton(
+        onPressed: selectedUserIds.isEmpty ? null : () {
+          final narrow = DmNarrow.withOtherUsers(
+            selectedUserIds,
+            selfUserId: store.selfUserId);
+          Navigator.pop(context);
+          Navigator.push(context,
+            MessageListPage.buildRoute(context: context, narrow: narrow));
+        },
+        child: Row(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            Text(zulipLocalizations.newDmSheetNextButtonLabel,
+              style: TextStyle(color: nextButtonColor,
+                fontSize: 20, height: 30 / 20)
+                .merge(weightVariableTextStyle(context, wght: 600))),
+            Icon(ZulipIcons.chevron_right,
+              size: 24,
+              color: nextButtonColor),
+          ])));
+  }
+}
+
+class _NewDmSearchBar extends StatelessWidget {
+  const _NewDmSearchBar({
+    required this.searchController,
+    required this.selectedUserIds
+  });
+
+  final TextEditingController searchController;
+  final Set<int> selectedUserIds;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+
+    return Container(
+      constraints: const BoxConstraints(
+        minHeight: 44,
+        maxHeight: 124),
+      padding: const EdgeInsets.symmetric(horizontal: 14, vertical: 11),
+      decoration: BoxDecoration(color: designVariables.bgSearchInput),
+      child: SingleChildScrollView(
+        reverse: true,
+        child: Row(children: [
+          Expanded(child: Wrap(
+            spacing: 6,
+            runSpacing: 4,
+            crossAxisAlignment: WrapCrossAlignment.center,
+            children: [
+              ...selectedUserIds.map((userId) => _SelectedUserChip(userId: userId)),
+              IntrinsicWidth(child: _buildSearchField(context)),
+            ])),
+        ])));
+  }
+
+  Widget _buildSearchField(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final hintText = selectedUserIds.isEmpty
+      ? zulipLocalizations.newDmSheetSearchHintEmpty
+      : zulipLocalizations.newDmSheetSearchHintSomeSelected;
+
+    return TextField(
+      controller: searchController,
+      cursorColor: designVariables.foreground,
+      style: TextStyle(color: designVariables.textMessage,
+        fontSize: 17, height: 22 / 17),
+      scrollPadding: EdgeInsets.zero,
+      decoration: InputDecoration(
+        isDense: true,
+        contentPadding: EdgeInsets.zero,
+        hintText: hintText,
+        hintStyle: TextStyle(color: designVariables.labelSearchPrompt,
+          fontSize: 17, height: 22 / 17),
+        border: InputBorder.none));
+  }
+}
+
+class _SelectedUserChip extends StatelessWidget {
+  const _SelectedUserChip({
+    required this.userId
```

Comment:
nit: this can be reduced to a single line, since the argument list is short enough